pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpressionUtils.java

```diff
@@ -155,10 +157,83 @@ private static RexExpression.FunctionCall handleSearch(RexCall rexCall) {
       return new RexExpression.FunctionCall(SqlKind.NOT_IN, dataType, SqlKind.NOT_IN.name(),
           toFunctionOperands(rexInputRef, sarg.rangeSet.complement().asRanges(), dataType));
     } else {
-      throw new NotImplementedException("Range is not implemented yet");
+      Set<Range<?>> ranges = sarg.rangeSet.asRanges();
+      return convertRangesToOr(dataType, rexInputRef, ranges);
     }
   }
 
+  private static RexExpression convertRangesToOr(ColumnDataType dataType, RexInputRef rexInputRef,
```

Comment:
i remember v1 have some optimizations on range predicates. i was wondering should we follow up with not converting it at plan time but instead do the conversion at runtime

Reply:
I think the correct solution would be to actually move these optimizations to planing time.

Reply:
Anyway, the issue we fixed here is that Calcite was creating an structure (a range) we don´t actually support. With this PR we generate the expected predicate (`field < xLiteral or field > yLiteral`) instead of the unsupported range.